src/MonoTorrent.PiecePicking/MonoTorrent.PiecePicking/StandardPieceRequester.cs

```diff
@@ -137,7 +135,7 @@ public void AddRequests (IRequester peer, ReadOnlyBitField available, ReadOnlySp
             if (!peer.IsChoking && peer.AmRequestingPiecesCount == 0) {
                 ReadOnlyBitField? filtered = null;
                 PieceSegment segment;
-                while (peer.AmRequestingPiecesCount < maxRequests) {
+                while (peer.CanRequestMorePieces) {
```

Comment:
This was what caused files to be picked out of order - effectively it always tried to add `MaxPendingRequests`, rather than relying on 'CanRequestMore'. Urgh